### PR TITLE
E2e/fix obw industry spec

### LINKED ISSUE #36259: E2E: Fix the `can save industry changes when navigating back to "Store Details"` test
### Describe the solution you'd like

Recently, the `can save industry changes when navigating back to "Store Details"` became flaky both in the nightly smoke test site and wp-env localhost. Notice from the Allure report below (generated on Dec. 30) that this test last passed on Dec 29th but started failing since Dec. 30.
Screen Shot 2023-01-02 at 7 14 06 PM

### PR DESCRIPTION
**Pull request by rodelgc**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?

### Changes proposed in this Pull Request:

Fixes recent flakiness in the `can save industry changes when navigating back to "Store Details"` E2E test by adding some assertions.

Closes #36259.

### How to test the changes in this Pull Request:

1. Go to https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml
2. Click on the "Run workflow" dropdown > select the branch of this PR (`e2e/fix-obw-industry-spec`) > then click "Run workflow. 
3. Wait for it to finish.
4. Verify that all the E2E tests passed:
    Screen Shot 2023-01-02 at 7 47 48 PM

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/complete-onboarding-wizard.spec.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ test.describe( 'Store owner can complete onboarding wizard', () => {
 			storeDetails.us.expectedIndustries
 		);
 		await page.click( 'button >> text=Continue' );
+		await expect( page ).toHaveURL( /.*step=product-types/ );
+		await expect(
+			page.locator( '.product-types button >> text=Continue' )
+		).toBeVisible();
```
Comment (rodelgc) [inline plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/complete-onboarding-wizard.spec.js:33]: These assertions make sure that the network calls triggered by clicking the "Continue" button are completed before moving on to the next test. This fixes the flakiness in the nightly smoke test site.

File: plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/complete-onboarding-wizard.spec.js [hunk 2]
```diff
@@ -41,8 +45,8 @@ test.describe( 'Store owner can complete onboarding wizard', () => {
 
 		// Navigate back to "Store Details" section
 		await page.click( 'button >> text=Store Details' );
-
 		await onboarding.handleSaveChangesModal( page, { saveChanges: true } );
+		await page.locator( 'text="Welcome to WooCommerce"' ).waitFor();
```
Comment (rodelgc) [inline plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/complete-onboarding-wizard.spec.js:49]: This ensures that the Store details page is loaded completely before navigating back to the Industry section. This fixes the flakiness in WP-Env localhost.

**Review by psealock — APPROVED** [review]
Tests passing and changes look good, thanks for handling @rodelgc

**Review by lanej0 — APPROVED** [review]
Looks good to me too!  Going to merge this fix.

**Pull request merged**